### DISCUSSION TITLE
is there any way to get __NEXT_DATA__ data in a seperate chunk / file.js ???

### DISCUSSION THREAD

**Discussion by tomer183** [vercel/next.js#34761] [Help]
i basicly use SSR with getInitialProps just to serve data to CDN (CloudFront),
is there any way to get __NEXT_DATA__ data in a seperate chunk / file.js ???

**Comment by Bahlaouane-Hamza** [comment]
Hi,
You will need to patch Next.js for this, here's one for next@12.1.6
You can use https://www.npmjs.com/package/patch-package to auto install the patch.

[next+12.1.6.patch](https://github.com/vercel/next.js/files/11009343/next%2B12.1.6.patch)

For this part
```js
+    delete props1.pageProps.node
+    delete props1.pageProps.i18n
+    delete props1.pageProps.menus
```
I'm removing only the props i don't wanna keep in the HTML, If you are using NextAuth for example, you want to keep `session` coming from SSR, and this may vary depending on your use case.

And finally on the _hydrate part, i just restore the props i removed.

Hope this helps.
